pkg/config/repoconfig/repoconfig.go

```diff
@@ -15,14 +15,43 @@ const (
 )
 
 type BotConfig struct {
-	LabelApproved *struct {
-		Approvals *int
-		Label     *string
-	} `yaml:"label-approved"`
+	LabelApproved *LabelApprovedConfig `yaml:"label-approved"`
 }
 
-func Read(gitRepo *git.Git, sha string) (*BotConfig, error) {
+type LabelApprovedConfig struct {
+	Approvals *int
+	Label     *string
+}
+
+func (c *BotConfig) HasLabelApproved() bool {
+	return c.LabelApproved != nil
+}
+
+func (c *BotConfig) GetLabelApproved() *LabelApprovedConfig {
+	if c.HasLabelApproved() {
+		return c.LabelApproved
+	}
+
+	return &LabelApprovedConfig{}
```

Comment:
```suggestion
	return nil
```

Safer like this, otherwise you're counting on caller to know to call `HasLabelApproved`

Reply:
May be we should panic instead, with a nicer explanation.

Otherwise we will end up in panic because of a nil de-reference.

I'll make it panic ;-)

Reply:
I didn't make it panic in the end, it would be weird to panic if somebody tried to do LabelApproved() == nil or something like that.

Reply:
Panic will cause bot to crash not a good idea IMHO :)